main.tf

```diff
@@ -1,3 +1,55 @@
-/********************************************************************
-This file is used to implement the ROOT module.
-*********************************************************************/
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_kms_vars = !var.skip_iam_authorization_policy && (var.kms_key_crn == null && var.existing_kms_instance_guid == null) ? tobool("When setting var.skip_iam_authorization_policy to true, a value must be passed for var.kms_key_crn and var.existing_kms_instance_guid") : true
+
+  # Determine what KMS service is being used for database encryption
+  kms_service = var.kms_key_crn != null ? (
+    can(regex(".*kms.*", var.kms_key_crn)) ? "kms" : (
+      can(regex(".*hs-crypto.*", var.kms_key_crn)) ? "hs-crypto" : null
+    )
+  ) : null
+}
+
+##############################################################################
+# AppID module
+##############################################################################
+
+resource "ibm_iam_authorization_policy" "policy" {
+  count                       = var.skip_iam_authorization_policy ? 0 : 1
```

Comment:
Your going to have to add another variable here because `skip_iam_authorization_policy` is not the only determining factor here. Can you be consistent with all other modules and add a `kms_encryption_enabled` variable

Reply:
`kms_encryption_enabled` needs to be used as well as part of this count logic